### DISCUSSION TITLE
How to remove an ORM class mapping from registry?

### DISCUSSION THREAD

**Discussion by saikiran2603** [sqlalchemy/sqlalchemy#11903] [Usage Questions] [answered]
We are creating ORM classes dynamically, and in a particular case , we end up updating the ORM class and re import the python module. In this case we get an error 

`sqlalchemy.exc.InvalidRequestError: Multiple classes found for path "<ClassName>" in the registry of this declarative base. Please use a fully module-qualified path.`

Since both the classes are essentially the same python file which gets re imported , how do we get rid of the older version of the class from registry ?

**Comment by CaselIT** [comment, marked as answer]
Hi,

IIRC the registry doesn't support removing orm classes, but if they go out of scope they will be gc-ed and removed by the registry.

It may be better to use a different registry when you recreate your classes so the new one will not conflict with the old ones.

**Reply by saikiran2603** [reply]
Thanks for this !!
